MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -72,29 +72,42 @@ double PiecewiseLinearInterpolation::getDerivative(
 
     auto const& it(std::lower_bound(_supp_pnts.begin(), _supp_pnts.end(),
                                     pnt_to_interpolate));
-    std::size_t const interval_idx = std::distance(_supp_pnts.begin(), it) - 1;
```

Comment:
@TomFischer I did this because when "pnt_to_interpolate==_supp_pnts.begin()", std::distance(_supp_pnts.begin(), it) returns zero, further minus 1 will lead to a very large value...